src/modules/fw_att_control/FixedwingAttitudeControl.cpp

```diff
@@ -381,7 +381,8 @@ void FixedwingAttitudeControl::Run()
 
 		/* lock integrator until control is started */
 		bool lock_integrator = !_vcontrol_mode.flag_control_rates_enabled
-				       || (_vehicle_status.vehicle_type == vehicle_status_s::VEHICLE_TYPE_ROTARY_WING && ! _vehicle_status.in_transition_mode);
+				       || (_vehicle_status.vehicle_type == vehicle_status_s::VEHICLE_TYPE_ROTARY_WING &&
+					   (!_vehicle_status.in_transition_mode || !_is_tailsitter));
```

Comment:
`(!_vehicle_status.in_transition_mode || !_is_tailsitter)`

change

`(!_vehicle_status.in_transition_mode && !_is_tailsitter)`

Reply:
I think how I propose it is correct, since we want to **not** lock the integrator in RW flight at all when being a tailsitter, resp. lock it in RW type and fixed-wing mode when not being a tailsitter.

Reply:
I do not quite understand.

eg: if we not in transition mode--->(!_vehicle_status.in_transition_mode || !_is_tailsitter)==true---> && (_vehicle_status.vehicle_type == vehicle_status_s::VEHICLE_TYPE_ROTARY_WING)-->true.This means that integral control will not be enabled in multi-rotor mode.

I think we should enable integral control not only during transmission, but also when flying in multi-rotor mode.

Reply:
`if (!ctl_data.lock_integrator && _k_i > 0.0f) {`

Reply:
> I think we should enable integral control not only during transmission, but also when flying in multi-rotor mode.

You're right, it should be a &&. Fixed it.